demos/kubeflow/README.md

```diff
@@ -0,0 +1,102 @@
+# E2E pipeline
+
+> This example is based on [canonical/kubeflow-example/e2e-wine-kfp-mlflow](https://github.com/canonical/kubeflow-examples/tree/main/e2e-wine-kfp-mlflow)
+> with some modifications to simplify the codebase. This example is explained
+> in more details in this [blog post](https://ubuntu.com/blog/mlops-pipeline-with-mlflow-seldon-core-and-kubeflow-pipelines).
+
+
+## Deploy kubeflow + mlflow
+
+```sh
+$ juju add-model kubeflow
+$ juju deploy kubeflow-lite --trust --debug
+
+# workaround for upstream issues in kubeflow:
+# https://github.com/kubeflow/manifests/issues/2087#issuecomment-1139260511
+$ sudo sysctl fs.inotify.max_user_instances=1280
+$ sudo sysctl fs.inotify.max_user_watches=655360
+```
+
+```sh
+# check the status of the deployment
+$ watch -c juju status --color --relations
+$ microk8s kubectl get all -n kubeflow
+```
+
+> :warning: you should wait for the pods to be "Running"
+
+```sh
+$ IPADDR=$(microk8s kubectl -nkubeflow get svc istio-ingressgateway-workload -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+$ juju config oidc-gatekeeper public-url=http://$IPADDR.nip.io
+$ juju config dex-auth public-url=http://$IPADDR.nip.io
+$ juju config dex-auth static-username=admin
+$ juju config dex-auth static-password=admin
+```
+
+The kubeflow dashboard will available at `$IPADDR.nip.io`
+
+To run this demo, we will also need mlflow.
+
+```sh
+$ juju deploy mlflow-server
+$ juju deploy charmed-osm-mariadb-k8s mlflow-db
+$ juju relate minio mlflow-server
```

Comment:
Reader for this document  may not familiar with juju.
We can add some command for them to verify the change or provide some comment. e.g.

* What is juju relate? ...

```sh
juju status -m model_name --relations
```